runtime/bindings/python/CMakeLists.txt

```diff
@@ -168,38 +168,36 @@ iree_py_test(
 # Install
 ################################################################################
 
-iree_py_install_package(
-  COMPONENT IreePythonPackage-runtime
-  PACKAGE_NAME iree_runtime
-  MODULE_PATH iree/runtime
-  DEPS
-    # TODO: Update CMake target/path mangling rules to make this syntactically
-    # rooted on "iree" in some way.
-    runtime_bindings_python_PyExtRt
-    iree_tools_iree-benchmark-module
-    iree_tools_iree-benchmark-trace
-    iree_tools_iree-run-module
-    iree_tools_iree-run-trace
-    ${_EXTRA_INSTALL_TOOL_TARGETS}
-  ADDL_PACKAGE_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/README.md
+set(_INSTALL_DIR "python_packages/iree_runtime")
+set(_INSTALL_COMPONENT "IreePythonPackage-runtime")
+
+# Install iree/runtime/*.py files verbatim into the tree.
+# We do this at the package level so as to avoid any loose files
+# from outside (i.e. tests, etc).
+install(
+  DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/iree/runtime/"
+  COMPONENT "${_INSTALL_COMPONENT}"
+  DESTINATION "${_INSTALL_DIR}/iree/runtime/"
+  FILES_MATCHING PATTERN "*.py"
 )
 
+# _runtime.so -> python_packages/iree_runtime/iree/_runtime.so
 install(
   # TODO: Update CMake target/path mangling rules to make this syntactically
   # rooted on "iree" in some way.
   TARGETS runtime_bindings_python_PyExtRt
-  COMPONENT ${PY_INSTALL_COMPONENT}
-  DESTINATION "${PY_INSTALL_MODULE_DIR}"
+  DESTINATION "${_INSTALL_DIR}/iree"
+  COMPONENT "${_INSTALL_COMPONENT}"
 )
 
+# Install tools into python_packages/iree_runtime/iree/runtime
 install(
   TARGETS
     iree_tools_iree-benchmark-module
     iree_tools_iree-benchmark-trace
     iree_tools_iree-run-module
     iree_tools_iree-run-trace
```

Comment:
FYI I tried to shuffle around the directory include order here https://github.com/google/iree/blob/66e48077386d347cf9113d15f332c8b14fd526bd/CMakeLists.txt#L693-L701

to include `runtime/` first, then `tools/`, but I ran into issues with this block of code (https://github.com/google/iree/pull/9084#discussion_r869493129)

```
CMake Error at runtime/bindings/python/CMakeLists.txt:196 (install):
  install TARGETS given target "iree_tools_iree-benchmark-module" which does
  not exist.
```

Reply:
Yeah: Most things in CMake don't have that problem, but unfortunately, install() targets must exist at the time of configure. Open to suggestions.

The nuclear option is: https://cmake.org/cmake/help/latest/command/cmake_language.html#deferring-calls

(could just defer these to the end of the project directory)

Reply:
My first thought was to have a section at the bottom of tools/CMakeLists.txt that would either directly set up the python installs, or include() a runtime/bindings/python/tools/CMakeLists.txt (that isn't included transitively by including runtime/) - basically deferred calls, but with code sprinkled across a few different files

Reply:
Actually, this might be an appropriate use of the nuclear option:

```
install(
  TARGETS
    iree_tools_iree-benchmark-module
    iree_tools_iree-benchmark-trace
    iree_tools_iree-run-module
    iree_tools_iree-run-trace
    ${_EXTRA_INSTALL_TOOL_TARGETS}
  DESTINATION "${_INSTALL_DIR}/iree/runtime"
  COMPONENT "${_INSTALL_COMPONENT}"
)
```

Becomes:

```
cmake_language(DEFER DIRECTORY "${IREE_SOURCE_DIR}"
  CALL install
  TARGETS
    iree_tools_iree-benchmark-module
    iree_tools_iree-benchmark-trace
    iree_tools_iree-run-module
    iree_tools_iree-run-trace
    ${_EXTRA_INSTALL_TOOL_TARGETS}
  DESTINATION "${_INSTALL_DIR}/iree/runtime"
  COMPONENT "${_INSTALL_COMPONENT}"
)
```

The cmake_language defer stuff is relatively terrifying. Someone seems to have been compelled to turn it into an RPC primitive... but this usage isn't too bad.